src/crimson/common/fixed_kv_node_layout.h

```diff
@@ -558,36 +559,35 @@ class FixedKVNodeLayout {
     set_meta(Meta::merge_from(left.get_meta(), right.get_meta()));
   }
 
-  static uint32_t get_balance_pivot_idx(
+  static std::optional<uint32_t> get_balance_pivot_idx(
     const FixedKVNodeLayout &left,
-    const FixedKVNodeLayout &right,
-    bool prefer_left)
+    const FixedKVNodeLayout &right)
   {
-    auto total = left.get_size() + right.get_size();
+    auto l_size = left.get_size();
+    auto r_size = right.get_size();
+    auto total = l_size + r_size;
     auto pivot_idx = total / 2;
-    if (total % 2 && prefer_left) {
-      pivot_idx++;
-    }
-    return pivot_idx;
+    return (pivot_idx == l_size || pivot_idx == r_size)
+      ? std::nullopt
```

Comment:
Is it really possible or reasonable hitting into this condition (nullopt) ?

See `fixed_tree::handle_merge()`, `merge_level()` is only called if the node is below min capacity (max-1)/2, this means, e.g.:
* cap = 6, min = 2, below-min = 1
* cap = 7, min = 3, below-min = 2

IIUC this means one of `left` and `right` must at below-min, then I don't understand why it is still possible to become `pivot_idx == l_size || pivot_idx == r_size`, because if the other node is at `min`, the node should be able to be merged instead, right ?

Reply:
There's a unit test case, `FixedKVNodeTest.balanced`, that can lead to this scenario. Should we remove that? Or keep the code? What do you think? @cyx1231st

Reply:
Remove or simplify according to the current discussions, I suppose.